lib/tic_tac_toe.rb

```diff
@@ -59,5 +59,41 @@ def board_full?
       !self.include?(:available)
     end
 
+    def best_move
+      move = cpu_check_for_wins_or_blocks(:player_two)
+      move ||= cpu_check_for_wins_or_blocks(:player_one)
+      move ||= cpu_take_open_space()
+      move
+    end
+
+    #CPU LOGIC METHODS
+    def open_spaces
+      spaces_available_index = self.each_index.select {|i| self[i] == :available}
+      spaces_available_index.map {|each| each}
+    end
+
+    def cpu_take_open_space
+      self.open_spaces().sample
+    end
+
+    def cpu_check_for_wins_or_blocks(player_symbol)
+      move = nil
+      for win_combo in WIN_COMBOS do
+        row = Array.new
+        for index in win_combo do
+          row.append(self[index])
+        end
+        if (row.count(player_symbol) == 2) && (row.include?(:available))
+          move = win_combo[row.index(:available)]
+        end
+      end
+      move
+    end
+
+
+
+
+
+
```

Comment:
instead of using a return value, why don't we just use an early return in the loop and then return nil at the end?
```suggestion
    def cpu_check_for_wins_or_blocks(player_symbol)
      for win_combo in WIN_COMBOS do
        row = Array.new
        for index in win_combo do
          row.append(self[index])
        end
        if (row.count(player_symbol) == 2) && (row.include?(:available))
          return win_combo[row.index(:available)]
        end
      end
      nil
    end
```